src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,121 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double DEFAULT_MULTIPLICATION_NUMBER = 1.5;
+    private static final int START_ARRAY_SIZE = 0;
+    private T[] elementsData;
+    private int capacity;
+    private int size;
+
+    @SuppressWarnings("unchecked")
+    public ArrayList() {
+        this.elementsData = (T[]) new Object[DEFAULT_CAPACITY];
+        this.capacity = DEFAULT_CAPACITY;
+        this.size = START_ARRAY_SIZE;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size() == capacity) {
+            double newCapacity = capacity * DEFAULT_MULTIPLICATION_NUMBER;
+            this.capacity = (int) newCapacity;
+            T[] newElementsData = (T[]) new Object[capacity];
+            System.arraycopy(elementsData, 0, newElementsData, 0, size);
+            elementsData = newElementsData;
```

Comment:
The logic for resizing the array is repeated in both add methods. Consider extracting this logic into a separate private method to adhere to the DRY principle.